src/walker/walker.ts

```diff
@@ -275,7 +275,14 @@ export class Walker extends EventEmitter<WalkerEmitter> {
         return [new ReferenceNode(fragment, '$ref is not a string'), fragment];
       } else if (walkingOptions.resolveRef !== null) {
         try {
-          fragment = walkingOptions.resolveRef(path, fragment.$ref);
+          let newFragment = walkingOptions.resolveRef(path, fragment.$ref);
+
+          if (typeof fragment.description === 'string') {
+            newFragment = { ...newFragment };
```

Comment:
what's the purpose of doing this clone here?

Reply:
We'd otherwise mutate the object, which would lead to a funky behavior in other parts of our ecosystem.

`walkingOptions.resolveRef` returns a portion of an object.

Example:

```js
const obj = {
   properties: {
     propA: {
        type: 'string',
     },
     propB: {
       $ref: '#/properties/propA',
       description: 'woop!',
    }
}
```

If we didn't create a new fragment, we'd end up overriding `obj.properties.propA` and the object would look.

```
const obj = {
   properties: {
     propA: {
        type: 'string',
        description: 'woop!',
     },
     propB: {
       $ref: '#/properties/propA',
       description: 'woop!',
    }
}
```

I hope that makes sense!

Reply:
Yes, thank you!